JamesNojekChallenge/Backend.py

```diff
@@ -0,0 +1,147 @@
+                  temp = temp.lstrip("\" ")
+                  temp = temp.rstrip("\"}")
+                  try:
+                    list.append(float(temp))
+                  except:
+                      return("Please ensure your file contains numbers only")
+              heapSort(list)            #sort and return
+              return(list)
+        except:
+              return("no data found, try posting")
+
+    def post(self):
+      try:
+        data = request.get_json()
+        if is_json(data):       #checks file type
+            f = open("data.json", "w")
+            text = json.dumps(data)
+            split = text.split(',')
+            if len(split) == 1:     #backup file check, if it is in list format it will be one long string with a length of 1
+                return("Please ensure your file is a JSON formatted list")
+            if len(split) != 500:   #number check
+                return("Please ensure your file is 500 numbers")
+            list = []
+            period = False
+            for k in range (0, 500):        #puts each number from JSON list into a new list of only floats
+                  tempList = split[k].split(':', 1)
+                  temp = tempList[1]
+                  temp = temp.lstrip("\" ")
+                  temp = temp.rstrip("\"}")
+                  try:
+                    list.append(float(temp))
+                  except:
+                      return("Please ensure your file contains numbers only")
+
+            f.write(text)
+            return("Data posted")       #success
+        else:
+            return("Please ensure your file is a JSON formatted list")
+      except:
+          return("Formatting error in JSON input")
+
+    def patch(self):
+        new = request.get_json()        #new number to patch in JSON format
+        text = json.dumps(new)
+        split = text.split(',')
+        list = []
+        if len(split) != 1:             #quantity check
+            return("Please only input 1 number")
+
+        tempList = split[0].split(':', 1)
+        temp = tempList[1]
+        temp = temp.lstrip("\" ")
+        temp = temp.rstrip("\"}")
+        try:
+            list.append(float(temp))
+            new = list[0]
+        except:
+            return("Please ensure your file contains numbers only")     #type check
+
+        sorted = Data.get(self)         #use get to create sorted list
+        i = 0
+        found = False
+        while found == False:           #find new placement
+            if new < sorted[0]:
+                sorted.insert(0,new)
+                return(sorted)
+            elif i + 1 == len(sorted):
+                sorted.append(new)
+                found = True
+            elif sorted[i + 1] < new:
+                i = i + 1
+            else:
+                sorted.insert(i + 1, new)
+                found = True
+                print("this2")
+        return(sorted)      #success
+
+
+api.add_resource(Data, '/data')

```

Comment:
This is a great way to build endpoints that have the same route. Without making any changes to the actual code, what would you do if I said I wanted each endpoint to have a different route? For example I want `GET /v1/data` and `PATCH /v2/data`

Reply:
I think you would have to create a different class, and call that class name to give it a different endpoint. If get and posted were still in the data class, you could have api.add_resource(Data, '/v1/data'). Then you could have a new class called DataV2 which contains the patch function definition, and then you'd have the endpoint as api.add_resource(DataV2, '/v2/data')